kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/NMDbusConnector.java

```diff
@@ -582,6 +602,18 @@ private void enableInterface(String deviceId, NetworkProperties properties, Devi
 
     }
 
+    private boolean isModemFailed(Device device, NMDeviceType deviceType) throws DBusException {
+        if (deviceType != NMDeviceType.NM_DEVICE_TYPE_MODEM) {
+            return false;
+        }
+        Optional<String> mmDbusPath = this.networkManager.getModemManagerDbusPath(device.getObjectPath());
+        if (!mmDbusPath.isPresent()) {
+            return false;
+        }
```

Comment:
I do not 100% agree on this.

This essentially means:
1. When Kura tried to activate a NM connection on a device, it threw an exception
2. while handling the exception Kura tries to retrieve the DBus path of the device that thew the exception
3. if it doesn't find the dbus path of the device it assumes the device is not in a "FAILED" state

It might be too optimistic. I would throw an exception in this case or log an error since we're in an unrecoverable state. What do you think?

Reply:
Indeed, I was unsure about this behavior. If the Dbus path is not present, we cannot say anything on the state of the modem. I returned a `false` to avoid the start of the timer. Probably, it is better to throw an exception and log. But we need to do the same when we start the `NMModemResetHandler` few lines below.